storage/range.go

```diff
@@ -774,6 +783,19 @@ func (r *Range) executeCmd(method string, args proto.Request, reply proto.Respon
 		return util.Errorf("unrecognized command %q", method)
 	}
 
+	// If we are applying a raft command, update the applied index.
```

Comment:
I noticed that if the method isn't read/write, the applied index won't be committed (as batch isn't written). I think perhaps we should combine these afterall.